Microsoft.Toolkit.Uwp.UI/ImageCache/ImageCache.cs

```diff
@@ -243,58 +237,47 @@ public static Task<BitmapImage> GetFromCacheAsync(Uri uri, bool throwOnError = f
+            var folder = await GetCacheFolderAsync();
+
+            var baseFile = await folder.TryGetItemAsync(key) as StorageFile;
+            if (await IsFileOutOfDate(baseFile, expirationDate))
+            {
+                baseFile = await folder.CreateFileAsync(key, CreationCollisionOption.ReplaceExisting);
+                try
                 {
-                    baseFile = await folder.CreateFileAsync(key, CreationCollisionOption.ReplaceExisting);
-                    try
+                    using (var webStream = await StreamHelper.GetHttpStreamAsync(uri))
                     {
-                        using (var webStream = await StreamHelper.GetHttpStreamAsync(uri))
+                        using (var reader = new DataReader(webStream))
                         {
-                            if (!isPreCache)
-                            {
-                                image = new BitmapImage();
-                                image.SetSource(webStream);
-                                webStream.Seek(0);
-                            }
-
-                            using (var reader = new DataReader(webStream))
-                            {
-                                await reader.LoadAsync((uint)webStream.Size);
-                                var buffer = new byte[(int)webStream.Size];
-                                reader.ReadBytes(buffer);
-                                await FileIO.WriteBytesAsync(baseFile, buffer);
-                            }
+                            await reader.LoadAsync((uint)webStream.Size);
+                            var buffer = new byte[(int)webStream.Size];
+                            reader.ReadBytes(buffer);
+                            await FileIO.WriteBytesAsync(baseFile, buffer);
                         }
                     }
-                    catch (Exception e)
-                    {
-                        await baseFile.DeleteAsync();
-                        throw e;
-                    }
                 }
-                else
+                catch (Exception)
                 {
-                    if (isPreCache)
-                    {
-                        return null;
-                    }
-
-                    using (var fileStream = await baseFile.OpenAsync(FileAccessMode.Read))
-                    {
-                        image = new BitmapImage();
-                        image.SetSource(fileStream);
-                    }
+                    await baseFile.DeleteAsync();
+                    throw; // rethrowing the exception changes the stack trace. just throw
                 }
+            }
 
-                if (MaxMemoryCacheSize > 0 && image != null)
-                {
-                    StoreToMemoryCache(key, image);
-                }
+            // the baseFile should be valid at this point and not null
+            using (var fileStream = await baseFile.OpenAsync(FileAccessMode.Read))
```

Comment:
Not trying to be a dick here but this undoes item 2 and 4 in #233 . The image is always read from the disk even if it was just downloaded and written to disk. In my opinion this is bad for performance reasons, since the webStream could be used to construct image. Lets say we have a list of 1000 images and the cache is empty, with the current implementation the result is a 1000 unnecessary disk reads.

Also PreCache has no advantage in this case, since image is always read from the disk even if we just want to ensure it is there. Lets say we would like to pre-cache 1000 images in a BackgroundWorker, with the current implementation we have 1000 unnecessary file reads and we consume unnecessary memory, because we are constructing 1000 BitmapImages that we don't even need.

The code might be simpler, but from perf point of view it is worse.